tests/unit/models/requests/test_requests.py

```diff
@@ -1,15 +1,71 @@
 from unittest import TestCase
 
-from xrpl.models.requests import Fee, GenericRequest
+from xrpl.models.exceptions import XRPLModelException
+from xrpl.models.requests import Fee, GenericRequest, Request
 
 
 class TestRequest(TestCase):
     def test_to_dict_includes_method_as_string(self):
-        tx = Fee()
-        value = tx.to_dict()["method"]
+        req = Fee()
+        value = req.to_dict()["method"]
         self.assertEqual(type(value), str)
 
     def test_generic_request_to_dict_sets_command_as_method(self):
         command = "validator_list_sites"
-        tx = GenericRequest(command=command).to_dict()
-        self.assertDictEqual(tx, {"method": command})
+        req = GenericRequest(command=command).to_dict()
+        self.assertDictEqual(req, {"method": command})
+
+    def test_from_dict(self):
+        req = {"method": "account_tx", "account": "rN6zcSynkRnf8zcgTVrRL8K7r4ovE7J4Zj"}
+        obj = Request.from_dict(req)
+        self.assertEqual(obj.__class__.__name__, "AccountTx")
+        expected = {**req, "binary": False, "forward": False}
+        self.assertDictEqual(obj.to_dict(), expected)
+
+    def test_from_dict_no_method(self):
+        req = {"account": "rN6zcSynkRnf8zcgTVrRL8K7r4ovE7J4Zj"}
+        with self.assertRaises(XRPLModelException):
+            Request.from_dict(req)
+
+    def test_from_dict_wrong_method(self):
+        req = {"method": "account_tx"}
+        with self.assertRaises(XRPLModelException):
+            Fee.from_dict(req)
+
+    def test_from_dict_noripple_check(self):
+        req = {
+            "method": "noripple_check",
+            "account": "rN6zcSynkRnf8zcgTVrRL8K7r4ovE7J4Zj",
+            "role": "user",
+        }
+        obj = Request.from_dict(req)
+        self.assertEqual(obj.__class__.__name__, "NoRippleCheck")
+        expected = {**req, "transactions": False, "limit": 300}
+        self.assertDictEqual(obj.to_dict(), expected)
+
+    def test_from_dict_account_nfts(self):
+        req = {
+            "method": "account_nfts",
+            "account": "rN6zcSynkRnf8zcgTVrRL8K7r4ovE7J4Zj",
+        }
+        obj = Request.from_dict(req)
+        self.assertEqual(obj.__class__.__name__, "AccountNFTs")
+        self.assertDictEqual(obj.to_dict(), req)
+
+    def test_from_dict_amm_info(self):
+        req = {
+            "method": "amm_info",
+            "amm_account": "rN6zcSynkRnf8zcgTVrRL8K7r4ovE7J4Zj",
+        }
+        obj = Request.from_dict(req)
+        self.assertEqual(obj.__class__.__name__, "AMMInfo")
+        self.assertDictEqual(obj.to_dict(), req)
+
+    def test_from_dict_generic_request(self):
+        req = {
+            "method": "tx_history",
+            "start": 0,
+        }
+        obj = Request.from_dict(req)
+        self.assertEqual(obj.__class__.__name__, "GenericRequest")
+        self.assertDictEqual(obj.to_dict(), req)
```

Comment:
These tests are not running as expected. The inputs are classified as `GenericRequest` instead of their specific class names.

```
    def test_from_dict_nft_history(self):
        req = {
            "method": "nft_history",
            "nft_id": "00080000B4F4AFC5FBCBD76873F18006173D2193467D3EE70000099B00000000",
        }
        obj = Request.from_dict(req)
        self.assertEqual(obj.__class__.__name__, "NFT_HISTORY")
        self.assertDictEqual(obj.to_dict(), req)

    def test_from_dict_nft_info(self):
        req = {
            "method": "nft_info",
            "nft_id": "00080000B4F4AFC5FBCBD76873F18006173D2193467D3EE70000099B00000000",
        }
        obj = Request.from_dict(req)
        self.assertEqual(obj.__class__.__name__, "NFT_INFO")
        self.assertDictEqual(obj.to_dict(), req)
```

Reply:
Good catch - another exception was needed.